massdash/structs/Data1D.py

```diff
@@ -122,23 +122,23 @@ def adjust_length(self, length):
             (new_data, new_intensity) : tuple of padded/truncated data and intensity
 
         """
-
         #### need to slice the array
         if length == len(self.data):
             new_data = self.data
             new_intensity = self.intensity
         elif length < len(self.data):
-            if length % 2 == 0:
-                slice_left = slice_right = length // 2
+            excess_length = len(self.data) - length
+            if excess_length % 2 == 0:
+                slice_left = slice_right = excess_length // 2
             else: # length % 2 == 1
-                slice_left = length // 2 + 1
-                slice_right = length // 2
+                slice_left = excess_length // 2
+                slice_right = excess_length + 1 // 2
```

Comment:
Does this give the expected results for `(excess_length + 1) // 2`?

Reply:
Seems like it, but I probably should add the parentheses to take higher precedence. :+1: